### PR TITLE
annotate heatmap with feature metadata (e.g., taxonomy)

### PR DESCRIPTION
**Pull request by nbokulich**
This will allow heatmaps to be labeled with feature metadata (e.g., taxonomy).

Contrast with taxa collapse: this will display individual ASVs annotated with whatever feature metadata you choose. So taxa are not collapsed.

### CODE REVIEW

File: q2_feature_table/tests/test_heatmap.py [hunk 1]
```diff
@@ -171,6 +194,57 @@ def test_munge_metadata_sort_samples(self):
...
+            pd.Series(['peanut', 'dog', 'cujo'], name='species',
+                      index=pd.Index(['O1', 'O2', 'O3'], name='id')))
+        obs = _munge_feature_metadata(feature_md, self.table, 'both')
+
+        exp = pd.DataFrame(
+            [[0, 10], [10, 12], [10, 11]], columns=['peanut', 'dog'],
+            index=pd.Index(['S1', 'S2', 'S3']))
+        assert_frame_equal(exp, obs)
+
+    def test_munge_feature_metadata_sort_samples(self):
```
Comment (thermokarst) [inline q2_feature_table/tests/test_heatmap.py]: Please add a test to cover the case when there are duplicate feature values (as described above).

File: q2_feature_table/_heatmap/_visualizer.py [hunk 2]
```diff
@@ -97,16 +97,32 @@ def _munge_metadata(metadata, table, cluster):
...
             normalize: bool = True, title: str = None,
             metric: str = 'euclidean', method: str = 'average',
             cluster: str = 'both', color_scheme: str = 'rocket') -> None:
     if table.empty:
         raise ValueError('Cannot visualize an empty table.')
 
-    if metadata is not None:
-        table = _munge_metadata(metadata, table, cluster)
+    if sample_metadata is not None:
+        table = _munge_metadata(sample_metadata, table, cluster)
```
[the remark this answers is not in the document]
Comment (nbokulich) [reply q2_feature_table/_heatmap/_visualizer.py]: 👍

File: q2_feature_table/_heatmap/_visualizer.py [hunk 3]
```diff
@@ -97,16 +97,32 @@ def _munge_metadata(metadata, table, cluster):
...
 
 
+def _munge_feature_metadata(metadata, table, cluster):
+    metadata = metadata.filter_ids(table.columns)
+    column_name = metadata.name
+    metadata_df = metadata.to_dataframe()
+    # replace feature IDs with feature metadata annotations
+    table.columns = metadata_df.reindex(table.columns)[column_name].values
+    if cluster == 'samples':
+        table = table.reindex(sorted(table.columns), axis=1)
```
[the remark this answers is not in the document]
Comment (nbokulich) [reply q2_feature_table/_heatmap/_visualizer.py]: good call, now using `sort_index` instead. Added a test for duplicate keys in new index.

File: q2_feature_table/tests/test_heatmap.py (hunk 1 above)
Comment (nbokulich) [reply q2_feature_table/tests/test_heatmap.py]: 👍

**Comment by nbokulich** [comment]
thanks @thermokarst ! I have addressed your comments in the most recent commit. 🏈
